thoth/storages/graph/postgres.py

```diff
@@ -6046,11 +6046,25 @@ def sync_solver_result(self, document: dict, *, force: bool = False) -> None:
                         "found in PythonPackageMetadata table, the error is not fatal"
                     )
 
+                if package_license["license"] == "UNDETECTED":
+                    package_license_name = None
+                    package_license_spdx = None
+
+                else:
+                    package_license_name = package_license["license"].get("full_name")
+                    package_license_spdx = package_license["license"].get("identifier_spdx")
+
+                if package_license["license_version"] == "UNDETECTED":
+                    package_license_version = None
+
+                else:
+                    package_license_version = package_license["license_version"]
+
                 license_metadata, _ = PythonPackageLicense.get_or_create(
                     session,
-                    license_name=package_license["license"].get("full_name"),
-                    license_identifier=package_license["license"].get("identifier_spdx"),
-                    license_version=package_license["license_version"],
+                    license_name=package_license_name,
+                    license_identifier=package_license_spdx,
+                    license_version=package_license_version,
```

Comment:
These field are non-nullable based on the scehma 
https://github.com/thoth-station/storages/blob/acd96e535a61e6f6d5d9ac857c38eef6de4fdaa8/thoth/storages/graph/models.py#L106

we should raise exception if the package_license info is not provided.

Reply:
I just took a look at spdx : https://scancode-licensedb.aboutcode.org/index.html.
Instead of None, I think we should refer to the spdx classification. See : 
- [no-license](https://scancode-licensedb.aboutcode.org/no-license.html)
- [unknown](https://scancode-licensedb.aboutcode.org/unknown.html)
- [unknown-spdx](https://scancode-licensedb.aboutcode.org/unknown-spdx.html)

Might be more.

Reply:
Is there any particular reason `python_package_license` are not nullable? Could we change the schema to allow them to be nullable, or would this require too many changes in license-solver code?

Reply:
I haven't investigated, but I'm not sure we should, given the alternatives of the mentioned "pseudo" spdx licences.

First, it makes us use an established standard.
Second, nullable types are always trickier to handle correctly (because that mean special casing unless you use a level of type-machinery I don't think Python)